### PR TITLE
Improves keyboard navigation accessibility

### LINKED ISSUE #72: Side navigation does not recieve keyboard focus
## Describe the bug
On pages with a side navigation, such as https://rockarch.org/collections/access-and-request-materials/, the menu items do not receive keyboard focus because the tabindex property is set to `-1`

## Expected behavior
The side navigation menu items should receive visible focus as part of the page's tab order.

## Impact on your work
The side navigation is not accessible using keyboard navigation

### PR DESCRIPTION
**Pull request by HaSistrunk**
- Changes strategy for main site navigation (using this [ARIA Authoring Practices example pattern](https://w3c.github.io/aria-practices/examples/disclosure/disclosure-navigation.html)) for better keyboard navigation options (using arrow keys, tab, and esc), and based on research on most commonly used a11y navigation patterns.
- Improves focus styles for keyboard navigation in main nav and side nav.
- Fixes #72 to add keyboard focus when not using a mobile device.
- Removes `inclusive-menu-button.js` and adjusts table of contents to use links instead of buttons

### CODE REVIEW

File: _includes/scripts.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <script src="https://code.jquery.com/jquery-3.3.1.min.js" crossorigin="anonymous"></script>
 
+<script src="{{ "/assets/js/disclosure-menu.min.js" | relative_url }}"></script>
 <script src="{{ "/assets/js/inclusive-menu-button.min.js" | relative_url }}"></script>
```
Comment (helrond) [inline _includes/scripts.html]: Does the script you added replace the previous JS (inclusive-menu-button.min.js)? If so we can get rid of this script import (and the script that's imported)

**Review by helrond — CHANGES_REQUESTED** [review]
Added one detailed comment but two other things:
1. when I've opened a menu using keyboard navigation, I am unable to close it and navigate to another top menu item. The spacebar scrolls, and the arrow keys seem boxed into the expanded menu items.
2. There's a weird styling thing happen with the open menus. I've tried flushing the cache, but to no avail.
Screen Shot 2020-05-05 at 9 55 38 AM

File: _includes/scripts.html (hunk 1 above)
Comment (HaSistrunk) [reply _includes/scripts.html]: The script I added replaces the inclusive-button.min.js for the top nav, but the side nav is still using it (and I think it is probably more appropriate for the side nav).

**Comment by HaSistrunk** [comment]
> when I've opened a menu using keyboard navigation, I am unable to close it and navigate to another top menu item. The spacebar scrolls, and the arrow keys seem boxed into the expanded menu items.

Yes, so I was reading through [the PR discussion for this pattern](https://github.com/w3c/aria-practices/pull/1036) leading to its addition as an example to the ARIA Authoring Practices. Originally there was no support for arrow keys, since the default for moving keyboard focus for screen reader use is `Tab / Shift + Tab`. However, the optional use of arrow keys to traverse open menu items was added for keyboard users who are not running a screen reader. `Esc` closes the menu and returns focus to the top-level button.

I can't find much info on user testing around these non-screenreader keyboard use patterns, but from my research on accessible link navigation with disclosure widgets, I like this pattern and it seems to be more widely used than using `role="menu"` for web page navigation (backed up by usa.gov, which uses a version of this without arrow key support).

**Comment by HaSistrunk** [comment]
I _think_ I fixed the weird styling issue with the open menu with my last commit. Thanks for catching that!

File: _includes/scripts.html (hunk 1 above)
Comment (helrond) [reply _includes/scripts.html]: Hmmm, I'm not a huge fan of scripts with overlapping functionality which serve different parts of the page. The more stuff we load on a page the slower it is, and the more possibilities there are for conflicts. Is there a way we can use one script for both purposes?

Comment (HaSistrunk) [reply _includes/scripts.html]: I think so, yes, but I was struggling some to understand how the side nav was working with the various scripts. I can revisit this.

Comment (helrond) [reply _includes/scripts.html]: Great, I want to make sure we're not unnecessarily creating technical debt.

**Comment by helrond** [comment]
This looks good to me! If we can get that issue with the scripts resolved I think this is good to go.

One thing I want to pay attention to on deployment is whether the new CSS styles get loaded in production without clearing the browser cache. If they don't we'll want to look at that and see if we can force that to happen somehow.

**Comment by HaSistrunk** [comment]
> This looks good to me! If we can get that issue with the scripts resolved I think this is good to go.
> 
> One thing I want to pay attention to on deployment is whether the new CSS styles get loaded in production without clearing the browser cache. If they don't we'll want to look at that and see if we can force that to happen somehow.

OK! I've removed the `inclusive-menu-button.js` script and updated the existing scripts to use links instead of buttons on the side nav, which simplifies that some.

**Review by helrond — APPROVED** [review]

**Pull request merged**